### PR TITLE
Toggle mute with left click while holding a modifier

### PR DESCRIPTION
**Pull request by andresmanz**
Patch for feature request #24. Not very clean, but it should work.

At least for once I _had_ to send a PR to the guy who made me love OSS.

### CODE REVIEW

**Comment by christophgysin** [comment]
Hi Andres

Thanks for your contribution. Good to see you're still participating in the OSS world!

**Comment by andresmanz** [comment]
That's good, I thought so while testing the changes, too. Personally, I prefer the alt key, but it doesn't really matter to me.

Of course, and I still enjoy learning by reading your code!

File: src/menu_info.c [hunk 1]
```diff
@@ -435,9 +435,16 @@ void menu_info_item_clicked(GtkWidget* item, GdkEventButton* event,
 
     switch(event->button)
     {
-        /* on left-click, set device as default */
+        /* on left-click, toggle mute if shift, alt or meta is pressed or set device as default */
```
Comment (christophgysin) [inline src/menu_info.c]: This comment is not valid anymore. To keep it simple, I would make individual comments for each action, explaining what button/modifier triggers it.

File: src/systray.c [hunk 2]
```diff
@@ -328,17 +328,24 @@ void systray_about_dialog()
...
 {
     menu_infos_t* mis = userdata;
+    menu_info_t* mi = &mis->menu_info[(ev->state & GDK_CONTROL_MASK) ? MENU_SOURCE : MENU_SINK];
+    menu_info_item_t* mii = menu_info_item_get_by_name(mi, mi->default_name);
 
     switch(ev->button)
     {
         case 1:
+            {
+                if((ev->state & GDK_MOD1_MASK) && mii)
```
Comment (christophgysin) [inline src/systray.c]: Is the check for mii != NULL necessary? I don't think the function should ever return NULL. If it ever does, let it crash so we can fix the root of the problem.

File: src/systray.c [hunk 3]
```diff
@@ -328,17 +328,24 @@ void systray_about_dialog()
 void systray_click_cb(GtkStatusIcon* icon, GdkEventButton* ev, gpointer userdata)
 {
     menu_infos_t* mis = userdata;
+    menu_info_t* mi = &mis->menu_info[(ev->state & GDK_CONTROL_MASK) ? MENU_SOURCE : MENU_SINK];
+    menu_info_item_t* mii = menu_info_item_get_by_name(mi, mi->default_name);
 
     switch(ev->button)
     {
         case 1:
+            {
```
Comment (christophgysin) [inline src/systray.c]: The extra block is not necessary, since you are not declaring any new variables in this case.

**Comment by andresmanz** [comment]
Thanks, done! I can't find your comment about NULL checks anymore, so I'll leave it like that for now.

**Comment by christophgysin** [comment]
It removed that comment again after realizing I was wrong.

Thanks again for your contribution!

**Comment by christophgysin** [comment]
merged in 379116e

**Pull request closed without merging**